internal/planner/planner.go

```diff
@@ -348,6 +348,10 @@ func getVersionConfigDiff(
 	strategy := config.RolloutStrategy
 	conflictToken := status.VersionConflictToken
 
+	if strategy.Strategy == temporaliov1alpha1.UpdateManual {
+		return nil
+	}
+
```

Comment:
when the user has specified Manual mode, I think we should never make changes to the routing config (even in the Current Version = nil case)